ekan0ra/queue.py

```diff
@@ -25,21 +25,22 @@ def enqueue(self, nick):
         """Add nick to queue."""
         self.append(nick)
 
-    def dequeue(self, nick):
+    def dequeue(self, nick, all=False):
         """
-        Remove first item matching `nick` in queue.
+        Remove every occurrence of `nick` from queue.
         """
-        try:
+        result=self.count(nick)>0
+        if all and result:
+            self = filter(lambda x: x != nick, self)
+        elif result:
             self.remove(nick)
-            return True
-        except ValueError:
-            return False
+        return result
 
     def has_next(self):
         """Check if queue has at least one item."""
         return len(self) > 0
 
-    def peek_next(self):
+    def peek(self):
```

Comment:
There is a call to this method here: [bot.py - Line 384](https://github.com/sayanchowdhury/theb0t/blob/master/ekan0ra/bot.py#L384)

That change should also be updated there.